src/utils/queue_builder.rs

```diff
@@ -70,3 +77,21 @@ pub fn queue_builder(
         }
     }
 }
+
+/// Sometimes, when the server hits a hard crash or a build thread panics,
+/// rustwide_builder won't actually remove the temporary directories it creates.
+/// Remove them now to avoid running out of disk space.
+fn remove_tempdirs() -> Result<(), io::Error> {
+    // NOTE: hardcodes that `tempfile::tempdir()` uses `std::env::temp_dir`.
+    for entry in std::fs::read_dir(std::env::temp_dir())? {
+        let entry = entry?;
+        if !Path::new(&entry.file_name()).starts_with(TEMPDIR_PREFIX) {
```

Comment:
How did you test this? 

It cloud be a platform difference, but I tried this part locally and no directory was deleted. 

From [the `starts_with`](https://doc.rust-lang.org/std/path/struct.Path.html#method.starts_with) docs it sounds like only full path components are compared, while here we are trying to match a part of the folder component, right?

Reply:
Wouldn't the comparison not just work on the `OsStr`?

Reply:
Oh uhh good point, it needs to be comparing the whole file path, not just the root name :sweat_smile: I didn't actually test this.

Reply:
How did you test this locally? Running `cratesfyi daemon` takes too long because it tries to add every crate on crates.io to the queue.

Reply:
`--disable-registry-watcher`

Reply:
ah right, I forgot I can add crates to the queue locally :laughing: